mteb/models/muq_mulan_model.py

```diff
@@ -0,0 +1,250 @@
+                            waveform = (
+                                audio.float()
+                            )  # assume it's already a torch.Tensor
+                        # Resample if needed
+                        if (
+                            item.get("sampling_rate", self.target_sampling_rate)
+                            != self.target_sampling_rate
+                        ):
+                            resampler = torchaudio.transforms.Resample(
+                                item["sampling_rate"], self.target_sampling_rate
+                            )
+                            waveform = resampler(waveform)
+                    else:
+                        continue
+
+                    # Add batch dimension and move to device
+                    wavs = waveform.unsqueeze(0).to(self.device)
+
+                    with torch.no_grad():
+                        audio_embeds = self.model(wavs=wavs)
+                        batch_features.append(audio_embeds.cpu().numpy())
+
+                all_features.extend(batch_features)
+
+            return np.vstack(all_features)
+        else:
+            # Process single batch
+            batch_features = []
+
+            for item in audio:
+                if isinstance(item, dict) and "array" in item:
+                    audio = item["array"]
+                    if isinstance(audio, np.ndarray):
+                        waveform = torch.from_numpy(audio).float()
+                    elif isinstance(audio, list):
+                        waveform = torch.tensor(audio, dtype=torch.float32)
+                    else:
+                        waveform = audio.float()  # assume it's already a torch.Tensor
+                    # Resample if needed
+                    if (
+                        item.get("sampling_rate", self.target_sampling_rate)
+                        != self.target_sampling_rate
+                    ):
+                        resampler = torchaudio.transforms.Resample(
+                            item["sampling_rate"], self.target_sampling_rate
+                        )
+                        waveform = resampler(waveform)
+                elif isinstance(item, torch.Tensor):
+                    waveform = item.float()
+                else:
+                    continue
+
+                # Add batch dimension and move to device
+                wavs = waveform.unsqueeze(0).to(self.device)
+
+                with torch.no_grad():
+                    audio_embeds = self.model(wavs=wavs)
+                    batch_features.append(audio_embeds.cpu().numpy())
+
+            return np.vstack(batch_features)
+
+    def get_text_embeddings(
+        self,
+        texts: list[str],
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Get text embeddings using MuQ-MuLan."""
+        with torch.no_grad():
+            text_embeds = self.model(texts=texts)
+
+        return text_embeds.cpu().numpy()
+
+    def encode(
```

Comment:
encode is really just for text embeddings. It has been confused and we should also fix the rest of the audio models separately.